src/LookingGlassXRWebGLLayer.ts

```diff
@@ -190,21 +220,22 @@ export default class LookingGlassXRWebGLLayer extends XRWebGLLayer {
 			gl.clearStencil(currentClearStencil)
 		}
 
-		const appCanvas = gl.canvas
-
 		let origWidth, origHeight
 
+
 		const blitTextureToDefaultFramebufferIfNeeded = () => {
 			if (!this[PRIVATE].LookingGlassEnabled) return
-
+			// if (cfg.viewCount) return
 			// Make sure the default framebuffer has the correct size (undo any resizing
 			// the host page did, and updating for the latest calibration value).
 			// But store off any resizing the host page DID do, so we can restore it on exit.
-			if (appCanvas.width !== cfg.calibration.screenW.value || appCanvas.height !== cfg.calibration.screenH.value) {
+
+			// check the dimensions of the canvas and ensure that we're not capturing
+			if ((appCanvas.width !== cfg.framebufferWidth || appCanvas.height !== cfg.framebufferHeight)) {
```

Comment:
The main change in the `WebXRWebGLLayer.ts` file and a key part of the refactoring, is drawing the main canvas at the full size of the frame buffer, then using the `drawImage` function of the canvas to draw the encoded image to the Looking Glass with respect for the square -> device aspect size. 

Previously we would change the dimensions of the main canvas to match the device resolution, which is not ideal for capturing quilts and breaks some resizing that webgl frameworks perform.